proposals/0370-modifiers.rst

```diff
@@ -292,8 +404,21 @@ Unresolved Questions
   to consider this idea separate from the quite considerable complexity of `#242`_,
   and so I have made it a separate proposal.
 
+.. _`#242`: https://github.com/ghc-proposals/ghc-proposals/pull/242
+
 * This proposal floats the idea of ``%oneShot`` and ``%inline``, but these
   might fit better as pragmas than modifiers. In any case, they are not
   proposed concretely here and would be subject to a future proposal.
 
-.. _`#242`: https://github.com/ghc-proposals/ghc-proposals/pull/242
+* How does this interact with Template Haskell?
+
+* What warning groups imply ``-Wunknown-modifiers``?
+
+* What scope are modifiers looked up in? For example::
+
+    %a data F a
+    %G data G
+
+  If these are accepted, they'd be meaningless under the current proposal. But
+  should they be rejected, or accepted with an unknow-modifiers warning, or
+  what?
```

Comment:
My current implementation allows using a modifier on its own definition:

```haskell
%CT1 data CT1

%CT2Con data CT2 = CT2Con

%CT2Con' data CT2' where
  CT2Con' :: CT2'

%CT3 class CT3 a
```

And has inconsistent behavior for free variables:

```haskell
-- accepted
%a data FV1 a

-- accepted
data FV2 a = %a FV2

-- rejected
data FV2' a where
  %a FV2' :: FV2'

-- accepted
data FV3 a = FV3 { fv3 %a :: () }

-- rejected
data FV3' a where
  FV3' :: { fv3' %a :: () } -> FV3

-- accepted, but seems especially sketchy
data FV4 = %a forall a . FV4 a

-- accepted
data FV4' where
  %a FV4' :: a -> FV4'

-- rejected
data FV5 = FV5 { fv5 %a :: forall a . a }

-- rejected
%a class FV6 a

-- rejected
%a instance B a
```

The easiest for me is to just continue not to think about this, and if in future it ever matters, we can decide then how it should behave. Downside is, at that point we might decide that programs currently accepted should be rejected. Maybe that's fine on grounds of "this is kind of experimental and no one has any reason to be writing those programs"?